clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -929,6 +938,9 @@ private void revokeAndAssign(SortedSet<TopicIdPartition> assignedTopicIdPartitio
                     String reason = interruptedReconciliationErrorMessage();
                     log.error("Interrupting reconciliation after partitions assigned callback " +
                         "completed. " + reason);
+                    if (memberHasRejoined) {
+                        attemptReconciliation = true;
```

Comment:
I guess this is trying to trigger a reconciliation if a member rejoins with the same assignment it was stuck reconciling? I would say we don't need to involve this new condition to reconcile here, and could simply let the `reconcile()` decide if it needs to reconcile when called on poll. We do need to fix the underlying problem, which is that we are discarding the reconciliation when a member rejoins with the same assignment ([KAFKA-16185](https://issues.apache.org/jira/browse/KAFKA-16185) to come in another PR). Once we got that right, I expect the reconciliation of the target received after rejoining would happen on poll if the target changed, or be reused when reconciliation completes, if the target is the same.

Reply:
The point was that we avoid the reconciliation-on-every-poll when metadata is missing by using `attemptReconciliation`, and I think we had to enable it here as well. However, since we decided that reconciliation-on-every-poll is fine for a short timeframe where metadata is incomplete, this does not matter anymore, I removed it.